File: app/src/main/java/com/bear2b/sampleapp/FCMService.java
package com.bear2b.sampleapp;

import android.util.Log;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;

public class FCMService extends FirebaseMessagingService {
    @Override
    public void onMessageReceived(RemoteMessage remoteMessage) {
        Log.e("DEV", "FCMService onMessageReceived: " + remoteMessage.toString());
        //TODO: handle message
    }
}

File: app/src/kotlin/java/com/bear2b/sampleapp/FCMService.kt
package com.bear2b.sampleapp

import android.util.Log
import com.bear.common.sdk.BearSdk
import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage

class FCMService : FirebaseMessagingService() {
    override fun onNewToken(token: String?) {
        super.onNewToken(token)
        token?.let { BearSdk.getInstance(this).registerFirebaseToken(it) }
    }

    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        Log.e("DEV", "FCMService onMessageReceived: $remoteMessage")
        //TODO: handle message
    }
}